rewards/classes/RewardsManager.py

```diff
@@ -219,22 +225,27 @@ def calculate_tree_distributions(self) -> RewardsList:
             self.chain,
         )
         console.log(
-            f"Fetched {len(tree_distributions)} tree distributions between {self.start} and {self.end}"
+            f"Fetched {len(tree_distributions)} "
+            f"tree distributions between {self.start} and {self.end}"
         )
         all_dist_rewards = []
         for dist in tree_distributions:
-            block = get_block_by_timestamp(self.chain, int(dist["timestamp"]))
+            start_block = get_block_by_timestamp(
+                self.chain, int(dist["end_of_previous_dist_timestamp"])
+            )
+            end_block = get_block_by_timestamp(self.chain, int(dist["timestamp"]))
             token = dist["token"]
             sett = dist["sett"]
-            # Dont blacklist tree rewards for emissions contracts
-            snapshot = self.fetch_sett_snapshot(block, sett, blacklist=False)
-            amount = int(dist["amount"])
-
-            cycle_logger.add_tree_distribution(sett, dist)
-            cycle_logger.add_sett_token_data(
-                sett, self.web3.toChecksumAddress(token), amount
+            snapshot = total_harvest_sett_snapshot(
+                self.chain,
+                start_block,
+                end_block,
+                sett,
+                blacklist=False,
+                number_of_snapshots=3,
```

Comment:
Magic number here - would this make sense to set as a config variable?

Reply:
Added constant